server/services/docs.ts

```diff
@@ -0,0 +1,99 @@
+import { readdirSync, readFileSync, statSync } from 'fs'
+import matter from 'gray-matter'
+import { nanoid } from 'nanoid'
+import { DocFileOrHeading, DocsMenuContent } from '../../types/interfaces'
+
+type DocsFrontMatter = {
+  priority?: number
+}
+
+const sortByPriority = (a: DocFileOrHeading, b: DocFileOrHeading): number => {
+  // As the highest priority is 1, we don't care about 0
+  if (a.priority && b.priority) return a.priority - b.priority
+  if (!a.priority && b.priority) return 1
+  if (a.priority && !b.priority) return -1
+  return 0
+}
+
+const convertFileOrDirNameToTitle = (str: string): string =>
+  str
+    .split('-')
+    .map((word) => `${word[0].toUpperCase()}${word.slice(1)}`)
+    .join(' ')
+
+const removeFileExtension = (fileOrDirName: string): string => {
+  const extensionIndex = fileOrDirName.indexOf('.')
+  return extensionIndex === -1 ? fileOrDirName : fileOrDirName.slice(0, extensionIndex)
+}
+
+const updateHeadingPriority = (
+  headingPriority: number | undefined,
+  childPriority: number | undefined
+): number | undefined => {
+  if (childPriority && childPriority > 0 && (!headingPriority || childPriority < headingPriority)) {
+    return childPriority
+  }
+  return headingPriority
+}
+
+const extractFrontMatterFromFile = (fileOrDirPath: string): DocsFrontMatter => {
+  const fileContents = readFileSync(fileOrDirPath).toString()
```

Comment:
Never use `readFileSync`, it will block all other threads from running until this file is read.  This will prevent other requests from being handled.  Instead you should be doing something similar to:

```
import { readFile } from 'fs/promises';

const fileContents = await readFile(fileOrDirPath, { encoding: 'utf8' })
```

Reply:
Done